build.gradle

```diff
@@ -212,9 +235,15 @@ dependencies {
   testImplementation group: 'io.rest-assured', name: 'json-path', version: versions.restAssured
   testImplementation group: 'io.rest-assured', name: 'xml-path', version: versions.restAssured
 
+  // groovy v3.* added to avoid 'WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass'
+  // in groovy:2.5.* when running functional/befta tests
+  testImplementation group: 'org.codehaus.groovy', name: 'groovy', version: '3.0.3'
```

Comment:
Just for my knowledge - where we use groovy?

Reply:
Not directly.

Even without befta (i.e. outside of this PR) the functional tests will generate warnings for groovy-2.5.11.

Adding `jvmArgs "--illegal-access=debug"` to the _functional_ task config generates full stack traces that point at _io.restassured_. If I increase the restAssured version to 4.3.0 (which is supposed to include the switchover to groovy 3.*) we now get only one warning (still referencing groovy 2.5.11) but we now also get an AbstractMethodError.  So decided to just bump groovy as it appears to resolve it.

Was tempted to use the dependencyManagement config to force the groovy version number but as the warnings only occurred during test but I didn't know If using DM would effect the libraries included in the main build.